target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom

```diff
@@ -131,6 +131,7 @@ case "$FIRMWARE" in
 "ath9k-eeprom-pci-0000:00:11.0.bin")
 	case $board in
 	buffalo,wzr-hp-ag300h|\
+	meraki,mr16|\
```

Comment:
ar71xx is actually reading from 0xff1000 here, you are reading from 0xfe1000. The equivalent is true for the other interface.
Despite, you don't set up a MAC address here manually, in contrast to ar71xx.

So, do you have correct MAC addresses on the WiFi interfaces right now?

Reply:
> ar71xx is actually reading from 0xff1000 here, you are reading from 0xfe1000. The equivalent is true for the other interface.

I mention this in the commit message, but was incorrect about which one ar71xx uses by default.

Meraki supplies four ARTs for each of two PHYs, all of which have correct checksums. Both sets of ARTs are functional but have slightly different properties (target powers in particular are higher in the upper 64KB). I was suspecting this was a change for different regions, as there are A and B models of the Meraki MR16 with otherwise identical hardware, but neither ART suggests a region.

Reply:
>if you remove the setup in DTS.

I'm sorry, I don't follow. The DTS must indicate where the ART partition is to protect it anyways. We could have this point directly to its location in firmware as you mention, but other models with AR922X and no separate EEPROM chip do not do this.

Reply:
> The value coming from the ART is available via /sys/kernel/debug/ieee80211/phy0/ath9k/base_eeprom, and is wrong and the same on all MR16's:

Well, I'd have just checked the flash at 0x1006, 0x5006, 0x11006 and 0x15006, but if that leads to the same result ...

> Meraki supplies four ARTs for each of two PHYs, all of which have correct checksums. Both sets of ARTs are functional but have slightly different properties (target powers in particular are higher in the upper 64KB). I was suspecting this was a change for different regions, as there are A and B models of the Meraki MR16 with otherwise identical hardware, but neither ART suggests a region.

Is there a GPL source or something where one could look what's actually used?

Reply:
Not quite. It's offset in the Modal EEPROM for each PHY as per the AR92XX document I would have attached if I could. It's since been removed from the Internet Archive.

I have no longer been able to find the MR16 source. It was released a while ago.

Reply:
> Not quite. It's offset in the base EEPROM for each PHY as per the AR92XX document I would have attached if I could. It's since been removed from the Internet Archive.

Ah, was mixing stuff up with ath10k a little.

Reply:
Since the vendor provided misconfigured ARTs I'm not going to do anything about this (most end users do not know how to modify AR92xx ARTs or even read them).